rabbitmq/main.go

```diff
@@ -0,0 +1,231 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/streadway/amqp"
+	"go.uber.org/zap"
+
+	"github.com/fission/keda-connectors/common"
+)
+
+type rabbitMQConnData struct {
+	queueName       string
+	url             string
+	includeUnacked  bool
+	fissionMetadata common.FissionMetadata
+}
+
+func parseRabbitMQConnData() (rabbitMQConnData, error) {
+	data := rabbitMQConnData{}
+	var err error
+	data.fissionMetadata, err = common.ParseFissionMetadata()
+	if err != nil {
+		return data, err
+	}
+
+	if os.Getenv("INCLUDE_UNACKED") != "true" {
+		data.includeUnacked = true
+	} else {
+		data.includeUnacked = false
+	}
+
+	data.queueName = os.Getenv("QUEUE_NAME")
+	if data.queueName == "" {
+		return data, fmt.Errorf("received empty queue name")
+	}
+
+	host := ""
+	if data.includeUnacked {
+		host = os.Getenv("HOST")
+	} else {
+		data.includeUnacked = true
+		host = os.Getenv("API_HOST")
+	}
+	if host == "" {
+		return data, fmt.Errorf("received empty host field")
+	}
+	data.url = host
+	return data, nil
+}
+
+func failOnError(logger *zap.Logger, err error, msg string) {
+	if err != nil {
+		logger.Fatal(msg, zap.Error(err))
+	}
+}
+
+func consumeMessage(data rabbitMQConnData, logger *zap.Logger, producerChannel, consumerChannel *amqp.Channel) {
+	msgs, err := consumerChannel.Consume(
+		data.queueName, // queue
+		"",             // consumer
+		false,          // auto-ack
+		false,          // exclusive
+		false,          // no-local
+		false,          // no-wait
+		nil,            // args
+	)
+	failOnError(logger, err, "failed while reading messages")
+
+	forever := make(chan bool)
+
+	go func() {
+		for d := range msgs {
+			success := handleFissionFunction(d, data, logger, producerChannel, consumerChannel)
+			if success {
+				d.Ack(false)
+			}
+		}
+	}()
+	logger.Info("Waiting for messages")
+	<-forever
+}
+
+func handleFissionFunction(d amqp.Delivery, data rabbitMQConnData, logger *zap.Logger, producerChannel, consumerChannel *amqp.Channel) bool {
```

Comment:
Let's rename to `handleHttpUrl` or something like that - in the interest of making if neutral as we discussed.

Reply:
Done